### ISSUE TITLE
Clarify documentation to indicate that Source constructors are deprecated

### ISSUE THREAD

**Issue by swizardlv** [mapbox/mapbox-gl-js#3050]
**mapbox-gl-js version**: v0.22.1

### Steps to Trigger Behavior

 1. index.js have code like 
```
var mapboxgl = require('mapbox-gl');
var geojsonsource = new mapboxgl.GeoJSONSource();
```
 2. in index.html  have a javascript link as bundle.js
 3.using webpack to build index.js to bundle.js following [webpack.config.example.js](https://github.com/mapbox/mapbox-gl-js/blob/master/webpack.config.example.js)

### Expected Behavior
no error in devtool console
### Actual Behavior
get an error in devtool console `mapboxgl.GeoJSONSource is not a constructor `

if I include the mapbox-gl.js in index.html directly, there will not be error.

Please give a hint to fix. Thanks.

**Comment by tmcw** [comment]
See the [upgrade notes with v0.22](https://github.com/mapbox/mapbox-gl-js/blob/master/CHANGELOG.md#breaking-changes) - mapboxgl.GeoJSONSource is removed, and you can use .addSource instead.

But keeping this open - [GeoJSONSource](https://www.mapbox.com/mapbox-gl-js/api/#GeoJSONSource) is still documented and should not be.

**Comment by lucaswoj** [comment]
> `GeoJSONSource` is still documented and should not be.

👍  The `GeoJSONSource` constructor has indeed been deprecated but `GeoJSONSource#setData` is still considered part of the public API. I think we can mark the constructor as private but keep the method.

**Comment by lucaswoj** [comment]
Fixed in db62cad58665f4554f11fd93a30e09e302fdbf89
